src/main/resources/com/tresys/pcap/xsd/pcap.dfdl.xsd

```diff
@@ -234,9 +235,13 @@ The message root is 'PCAP'.
 
   <xs:element name="NetworkLayer">
     <xs:complexType>
-      <xs:choice>
-        <xs:element ref="pcap:IPv4" />
-        <xs:element ref="pcap:IPv6" />
+      <xs:choice dfdl:choiceDispatchKey="{
+       if (xs:unsignedInt(../Ethertype) eq 2048) then 'ipV4'
+       else  if (xs:unsignedInt(../Ethertype) eq 34525) then 'ipV6'
+       else fn:error('unrecognized Network Layer type')
+       }">
```

Comment:
The if-else-if is going to get ugly as more network layers are eventually added. Thoughts on just dispatching on ``xs:string(../Ethertype)`` and the choiceBranchKey's become "2048" and "34525"? I guess we would lose the fn:error, but I would think choice dispatch error for a unknown key would still be helpful.   Or we could do the choice of choice trick to have an error (or really,  a choiceBranchDefault would be really helpful here).

Reply:
I will remove all the if-then-else I can. I agree they're just clutter.